rawfile/bd2fs.py

```diff
@@ -284,22 +289,36 @@ def ControllerExpandVolume(self, request, context):
     @log_grpc_request
     def CreateSnapshot(self, request: csi_pb2.CreateSnapshotRequest, context):
         with VolLock(request.source_volume_id):
-            file = img_file(request.source_volume_id)
-            loop_dev = attach_loop(file)
-            fs = from_device(loop_dev)
-            if not fs:
-                raise UnknownFileSystemError(
-                    device=loop_dev, volume_id=request.source_volume_id
+            volume_meta = metadata(request.source_volume_id)
+            freezefs = volume_meta.get("freezefs", False)
+            copy_on_write = volume_meta.get("copy_on_write", False)
+            current = snapshot_manager.list_snapshots(
+                volume_id=request.source_volume_id, snapshot_name=request.name
+            )
+            nano = 10**9
+            if current:
+                snapshot = current[0]
+                return csi_pb2.CreateSnapshotResponse(
+                    snapshot=csi_pb2.Snapshot(
+                        size_bytes=snapshot.size_bytes,
+                        snapshot_id=f"{snapshot.volume_id}/{snapshot.name}",
+                        source_volume_id=snapshot.volume_id,
+                        ready_to_use=snapshot.ready,
+                        creation_time=Timestamp(
+                            seconds=int(snapshot.creation_time),
```

Comment:
this is implemented differently below

Reply:
This is for handling snapshot creation in background, for progress tracking external snapshotter sends multiple snapshot create multiple times, I will check that if the snapshot already exists I will return the existing one's information